packages/state-transition/src/cache/epochTransitionCache.ts

```diff
@@ -200,21 +221,41 @@ export function beforeProcessEpoch(
   const validators = state.validators.getAllReadonlyValues();
   const validatorCount = validators.length;
 
+  // pre-fill with true (most validators are active)
+  isActivePrevEpoch.length = validatorCount;
+  isActiveCurrEpoch.length = validatorCount;
+  isActiveNextEpoch.length = validatorCount;
+  isActivePrevEpoch.fill(true);
+  isActiveCurrEpoch.fill(true);
+  isActiveNextEpoch.fill(true);
+
+  // During the epoch transition, additional data is precomputed to avoid traversing any state a second
+  // time. Attestations are a big part of this, and each validator has a "status" to represent its
+  // precomputed participation.
+  // - proposerIndex: number; // -1 when not included by any proposer
+  // - inclusionDelay: number;
+  // - flags: number; // bitfield of AttesterFlags
+  proposerIndices.length = validatorCount;
+  inclusionDelays.length = validatorCount;
+  flags.length = validatorCount;
```

Comment:
nice technique on increasing the length, we don't need the `fill()` method below if we always set them at every index

Reply:
fill is being used there to (cheaply) reset the values